server/hwapi/external/c3/client.py

```diff
@@ -44,6 +47,77 @@ class C3Client:
 
     def __init__(self, db: Session):
         self.db = db
+        self.session = self._create_session_with_retries()
+
+    def _create_session_with_retries(self):
+        """Create a requests session with retry strategy"""
+        session = requests.Session()
+
+        retry_strategy = Retry(
+            total=5,  # Total number of retries
+            backoff_factor=2,  # Wait time between retries (2**retry_count seconds)
+            status_forcelist=[429, 500, 502, 503, 504],  # HTTP status codes to retry
+            allowed_methods=["HEAD", "GET", "OPTIONS"],  # Only retry safe methods
+            raise_on_status=False,  # Don't raise exception on retry-able status codes
+        )
+
+        # Mount adapter with retry strategy
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        session.mount("http://", adapter)
+        session.mount("https://", adapter)
+
+        return session
+
+    def _make_request_with_retries(
+        self, url: str, max_retries=5, base_delay=2, max_delay=60
+    ):
+        """Make HTTP request with custom retry logic for timeout errors"""
+        for attempt in range(max_retries):
+            try:
+                logger.debug(
+                    f"Attempting request to {url} (attempt {attempt + 1}/{max_retries})"
+                )
```

Comment:
It's recommended to not form the string in the log calls for (somewhat minor) efficiency gains:

```suggestion
                logger.debug(
                    "Attempting request to %s (attempt %d/%d)", url, attempt + 1, max_retries
                )
```